src/compat/posix/include/sys/mman.h

```diff
@@ -28,27 +30,20 @@ __BEGIN_DECLS
 
 #define MAP_FAILED (void*)-1
 
-#if 1
-#include <errno.h>
-static inline void  *mmap(void *addr, size_t len, int prot, int flags, int fd, off_t off) {
-	// ToDo: implement for InitFS files
-	(void)addr;
-	(void)len;
-	(void)prot;
-	(void)flags;
-	(void)fd;
-	(void)off;
-	//printf(">>> mmap(%i)\n",fd);
-	errno = EPERM;
-	return NULL;
-}
-
+extern struct page_allocator *__phymem_allocator;
+extern int    mprotect(void *, size_t, int);
 
+// todo: implement for InitFS files
+static inline void  *mmap(void *addr, size_t len, int prot, int flags, int fd, off_t off) {
+	size_t full_pages = len < PAGE_SIZE() ? 1 : len / PAGE_SIZE();
+	size_t pages = len % full_pages ? full_pages + 1 : full_pages;
```

Comment:
You, probably, meant 
`size_t pages = len % (full_pages ** * PAGE_SIZE() **) ? full_pages + 1 : full_pages;`

You can shorten the whole expression to
`int pages = (len + PAGE_SIZE() - 1) / PAGE_SIZE();`